testing/acceptance_tests/circuit_test.py

```diff
@@ -0,0 +1,695 @@
+import pytest
+import os
+import sys
+import json
+import numpy as np
+import copy
+
+from zquantum.core import circuit
+from zquantum.core.utils import RNDSEED
+from zquantum.core.circuit import (
+    load_circuit_template_params,
+    save_circuit_template_params,
+    load_circuit,
+    save_circuit,
+    save_circuit_set,
+    load_circuit_set,
+    load_parameter_grid,
+    load_circuit_layers,
+    load_circuit_connectivity,
+    Circuit,
+)
+from zquantum.core.testing import create_random_circuit as _create_random_circuit
+
+sys.path.append("../..")
+from steps.circuit import (
+    build_ansatz_circuit,
+    create_random_circuit,
+    generate_random_ansatz_params,
+    combine_ansatz_params,
+    build_uniform_param_grid,
+    build_circuit_layers_and_connectivity,
+    add_ancilla_register_to_circuit,
+    concatenate_circuits,
+    batch_circuits,
+)
+
+
+class Test_generate_random_ansatz_params:
+    @pytest.mark.parametrize(
+        "number_of_layers, problem_size",
+        [
+            (1, 1),
+            (0, 1),
+            (1, 6),
+            (4, 3),
+        ],
+    )
+    def test_generate_random_ansatz_params_using_mock_ansatz_specs(
+        self, number_of_layers, problem_size
+    ):
+        # Given
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": number_of_layers,
+            "problem_size": problem_size,
+        }
+        seed = RNDSEED
+
+        filename = "params.json"
+        if os.path.exists(filename):
+            os.remove(filename)
+
+        # When
+        generate_random_ansatz_params(ansatz_specs=ansatz_specs, seed=seed)
+
+        # Then
+        assert os.path.exists(filename)
+        parameters = load_circuit_template_params(filename)
```

Comment:
The thing we're doing here is ridiculous.
Steps really should be python functions that return python objects.
And these python objects should be serialized in the runtime (or something).
Sorry...
It's not the place nor time for such discussion...

Reply:
Agreed - I might take some time tomorrow for a small innovation day to see if we can't add something simple to the runtime to handle this

Reply:
Maybe the first step would be something simple where you can return basic types and the runtime will serialize it for you. Then we can build off of that to add support for our types. Because all we need is a way to make a map from type -> serialization method. The question just then becomes how to make that map update when you have different resources installed